network-api/networkapi/wagtailpages/utils.py

```diff
@@ -86,3 +90,47 @@ def get_mini_side_nav_data(context, page, no_minimum_page_count=False):
         'current': page,
         'menu_pages': menu_pages,
     }
+
+
+def get_content_related_by_tag(page, result_count=3):
+    """
+    Get all posts that feel related to this page, based
+    on its `.tags` content. If it has tags.
+    """
+
+    tagged_post_types = (
+        apps.get_model('wagtailpages', 'BlogPage'),
+        # Add more models here as we add tags to more types
+    )
+
+    own_tags = page.tags.all()
+
+    related_qs = False
+
+    for post_type in tagged_post_types:
+        related_posts = post_type.objects.filter(tags__in=own_tags).distinct()
+
+        # Exlude "this page" from the result set for the page's own.
+        # page type, so we don't yield "this page is similar to itself".
+        if page.__class__ is post_type:
+            related_posts = related_posts.exclude(pk=page.pk)
+
+        if related_qs is False:
+            related_qs = related_posts
+        else:
+            related_qs = related_qs | related_posts
```

Comment:
What's this pipe doing? Appending?

Reply:
Technically it forms an `OR`d queryset, but yes effectively it's an append: https://docs.djangoproject.com/en/2.1/ref/models/querysets/#or